lib/github-changelog.php

```diff
@@ -24,9 +31,69 @@ function make_github_request( $url, $headers = array() ) {
 	return json_decode( $data, true );
 }
 
+/**
+ * Wrapper which paginates the GitHub request to fetch everything. Useful when
+ * the endpoint doesn't include pagination information.
+ *
+ * @param string $url URL to get
+ * @return array $data The array with all the data
+ */
+function make_github_request_paginated( $url ) {
+	$all_data       = array();
+	$pagination_url = $url . '?per_page=50&page=';
+
+	for ( $page = 1; ; $page++ ) {
+		$data = make_github_request( $pagination_url . $page );
+		if ( empty( $data ) ) {
+			break;
+		}
+		$all_data = array_merge( $all_data, $data );
+	}
+
+	return $all_data;
+}
+
+/**
+ * Get the PRs referenced in the commits of a PR.
+ *
+ * @param array $pr The PR object from GitHub API
+ * @return array The PR objects from PR IDs referenced in the commits
+ */
+function get_referenced_prs( $pr ) {
+	if ( ! isset( $pr['_links']['commits']['href'] ) ) {
+		return array();
+	}
+
+	$commits = make_github_request_paginated( $pr['_links']['commits']['href'] );
+	$pr_ids  = array();
+
+	foreach ( $commits as $commit ) {
+		$msg = $commit['commit']['message'];
+
+		echo "Checking commit: {$commit['sha']}\n";
```

Comment:
Is this output intentional, or did we forget it here? :D

Reply:
https://github.com/Automattic/vip-build-tools/pull/15/commits/a96ebdee62f299c39b6e34abce6c34ade5d92d7a